server/api/categoryRouter.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * Router for item actions
+ */
+const Category = require('../db/models/category');
+const express = require('express');
+const Item = require('../db/models/item');
+const router = express();
+router.use(express.json());
+
+/**
+ * Create a new category
+ * 
+ * req.body: {title: String}
+ * 
+ * response: {success: boolean, message: String}
+ */
+router.post('/createCategory', (req, res, next) => {
+  const newCategory = req.body;
+  let response = {};
+  Category.create(newCategory, (err, data) => {
+    if (err) {
+      response.successs = false;
+      response.message = 'failed to add category';
+      next(err);
+    } else {
+      response.success = true;
+      response.message = 'created new category';
+    }
+    res.json(response);
+  });
+});
+
+/**
+ * Get all of the user's categories- just names and ids.
+ * 
+ * req.body: {userId: mongoose.objectId}
+ * 
+ * response: [{ title: String, _id: Mongoose.objectId}]
+ */
+router.get('/getCategories', async (req, res, next) => {
+  try {
+    const userId = req.body.userId;
+    const categoriesData = await Category.find({ userId: userId }, 'title', function (err, data) {
+      if (err) {
+        next(err);
+      }
+    });
+    const response = categoriesData;
+    res.json(response);
+  } catch (err) {
+    next(err);
+  }
+});
+
+/**
+ * Get all the items belonging in a specified category
+ * 
+ * req.body: {categoryId: mongoose.objectId}
+ * 
+ * response: [{title: String, 
+ *             items: [{
+ *               item objects from db 
+ *             }]
+ *            }]
+ */
+router.get('/getCategoryItems', async (req, res, next) => {
+  const categoryId = req.body.categoryId;
+  let response = {};
```

Comment:
This code should also go into the try block

Reply:
okay